src/Type/UnionType.php

```diff
@@ -117,7 +136,7 @@ public function getConstantArrays(): array
 
 	public function getConstantStrings(): array
 	{
-		return UnionTypeHelper::getConstantStrings($this->getTypes());
+		return UnionTypeHelper::getConstantStrings($this->getTypesOfClass(ConstantStringType::class));
```

Comment:
@herndlm do you think we have a similar problem for other methods in UnionType calling into UnionTypeHelper?

Reply:
pretty much looks like that I guess :/ there's a difference at least. there might be another though that worried me already a bit in the past: the old `TypeUtils::map()` was often called with `$inspectIntersections` set to `false`, which would do exactly that, not inspect intersections, but I think inspecting them made things better / stricter, so maybe that one's fine

Reply:
tbh I also don't like the "stop on unmatched" logic though, it would also be fixable by e.g. ensuring that `isConstantArray()->yes()` on calling site, right? hmm. sorry, still just thinking out loud here

Reply:
I think its only relevant for the cases where the 4th arg to `map` is `true`.

e.g. for these array-methods this is not the case

https://github.com/phpstan/phpstan-src/blob/4a72d1513c86df6aab55672bd676c1fcb8c03f9b/src/Type/TypeUtils.php#L120-L136